core/libp2p/multi/multibase.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_MULTIBASE_HPP
+#define KAGOME_MULTIBASE_HPP
+
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "common/buffer.hpp"
+#include "common/result.hpp"
+
+namespace libp2p::multi {
+  /**
+   * Allows to distinguish between different base-encoded binaries
+   * See more: https://github.com/multiformats/multibase
+   */
+  class Multibase {
```

Comment:
maybe `MultibaseCodec` ?

Reply:
I wanted to keep the "external" names - those, which are from Libp2p specification - exactly as they are in the docs, so that, for example, process of contribution by some other people would be easier

Reply:
Did you check what miltibase object does in other implementations?

Reply:
For example, in [Go](https://github.com/multiformats/go-multibase/blob/master/multibase.go) it does exactly the same, having two methods for encoding and decoding